src/matches.rs

```diff
@@ -386,14 +392,17 @@ fn rewrite_match_body(
         let comment_str = arrow_snippet[arrow_index + 2..].trim();
         if comment_str.is_empty() {
             String::new()
+        } else if (!was_block && !is_block) || is_empty_block {
```

Comment:
Which case in the test of the issue is covered by `!was_block && !is_block` ?

Reply:
It is case 1, when we have single statement match arm with arrow-comment.

Reply:
Thanks